app/src/main/java/com/muzima/utils/smartcard/KenyaEmrShrMapper.java

```diff
@@ -0,0 +1,1355 @@
+        } catch (Throwable E){
+            Log.e(TAG, "Could not create and save registration payload");
+        }
+    }
+
+    public static void updatePatientDemographicsWithCardSerialNumberAsIdentifier(MuzimaApplication muzimaApplication, final Patient patient, final String cardSerialNumber){
+        try {
+            Location defaultLocation = LocationUtils.getDefaultEncounterLocationPreference(muzimaApplication);
+            if(defaultLocation != null) {
+                LocationAttribute attribute= defaultLocation.getAttribute(Constants.Shr.KenyaEmr.LocationAttributeType.MASTER_FACILITY_CODE.name);
+                String facilityMflCode = attribute.getAttribute();
+                PatientIdentifier patientIdentifier = PatientIdentifierUtils.getOrCreateKenyaEmrIdentifier(muzimaApplication,
+                        cardSerialNumber, Constants.Shr.KenyaEmr.PersonIdentifierType.CARD_SERIAL_NUMBER.shr_name,
+                        facilityMflCode);
+                patient.addIdentifier(patientIdentifier);
+                muzimaApplication.getPatientController().updatePatient(patient);
+
+                //ToDo: Create demographics update payload for card serial number
+                FormData formData = new FormData() {{
+                    setUuid(UUID.randomUUID().toString());
+                    setPatientUuid(patient.getUuid());
+                    setUserUuid("userUuid");
+                    setStatus(STATUS_COMPLETE);
+                    setTemplateUuid(Constants.Shr.KenyaEmr.REGISTRATION.FORM.FORM_UUID);
+                    setDiscriminator(FORM_JSON_DISCRIMINATOR_SHR_DEMOGRAPHICS_UPDATE);
+                }};
+
+                User user = muzimaApplication.getAuthenticatedUser();
+
+                JSONObject formDataJSON = new JSONObject(new HTMLPatientJSONMapper().map(patient, formData, user, true));
+                JSONObject demographicsUpdateJson = new JSONObject();
+
+                JSONObject newIdentifierJson = new JSONObject();
+                newIdentifierJson.put("identifier_type_name", PersonIdentifierType.CARD_SERIAL_NUMBER.name);
+                newIdentifierJson.put("identifier_type_uuid", PersonIdentifierType.CARD_SERIAL_NUMBER.uuid);
+                newIdentifierJson.put("identifier_value", cardSerialNumber);
+
+                demographicsUpdateJson.put("demographicsupdate.otheridentifier",newIdentifierJson);
+                formDataJSON.put("demographicsupdate", demographicsUpdateJson);
+
+                formData.setJsonPayload(formDataJSON.toString());
+                muzimaApplication.getFormController().saveFormData(formData);
+            }
+        } catch (Throwable e){
+            Log.e(TAG, "Error updating patient identifier. ",e);
+        }
+    }
+
+    /**
+     *
+     * @param shrModel
+     * @return
+     */
+    public static List<PersonName> extractPatientNamesFromShrModel(KenyaEmrShrModel shrModel) throws ShrParseException {
```

Comment:
functions with shrModel should be renamed to SHRModel for consistency in naming.